spec/models/login_spec.rb

```diff
@@ -3,7 +3,19 @@
     expect(subject).to belong_to(:account).with_foreign_key(:user_id)
   end
 
-  it { is_expected.to validate_presence_of(:identification) }
+  subject { Login.new(identification: 'test@example.com', oauth2_token: 'token', single_use_oauth2_token: 'oldtoken') }
+
+  describe 'Login model validations' do
+    before do
+      Login.any_instance.stub(:ensure_oauth2_token).and_return(true)
+      Login.any_instance.stub(:assign_single_use_oauth2_token).and_return(true)
+    end
```

Comment:
I think these mocks should enable adding specs for the presence validations of `oauth2_token` and `single_use_oauth2_token`? Can you add those below?